test/libsolidity/syntaxTests/nameAndTypeResolution/042_functions_with_stucts_of_non_external_types_in_interface_2.sol

```diff
@@ -5,4 +5,5 @@ contract C {
     function f(S memory) public {}
 }
 // ----
-// TypeError: (105-113): Only libraries are allowed to use the mapping type in public or external functions.
+// TypeError: (105-113): Types containing (nested) mappings can only be used in storage.
+// TypeError: (105-113): Only libraries are allowed to use a (nested) mapping type in public or external functions.
```

Comment:
The error text might be misleading - the critical detail is whether the data location is storage or not.

Reply:
Maybe we can just remove that check?